src/components/Onboarding/Onboarding.tsx

```diff
@@ -1,75 +1,104 @@
 import React, { useContext, useEffect, useState } from 'react';
+import { API, Auth } from 'aws-amplify';
+import { createStackNavigator } from '@react-navigation/stack';
 import { AppContext } from '../../contexts/AppContext';
+import { homeInfo } from './onboardingData';
 import Page1 from './OnboardingPage1';
 import Page2 from './OnboardingPage2';
 import Page3 from './OnboardingPage3';
 import Page4 from './OnboardingPage4';
+import {
+  createHome,
+  createHomeOwner,
+  updateHome,
+} from '../../graphql/mutations';
+import { CreateHomeInput } from '../../API';
+import { getUser } from '../../graphql/queries';
+import { UserContext } from '../../contexts/UserContext';
 
-const Onboarding: React.FC = () => {
-  const [page, setPage] = useState('page1');
-  const [homeData, setHomeData] = useState({
-    AC: '',
-    EV: '',
-    address: '',
-    bathrooms: '',
-    bedrooms: '',
-    city: '',
-    electricEnergyUsage: '',
-    fuelType: '',
-    gasUsage: '',
-    heaterAge: '',
-    hotTub: '',
-    outdoor: '',
-    pool: '',
-    squareFeet: '',
-    state: '',
-    waterFuelType: '',
-    waterHeaterAge: '',
-    waterUse: '',
-    year: '',
-    zipCode: '',
-  });
+const Stack = createStackNavigator();
+
+interface Props {
+  homeInformation?: CreateHomeInput;
+}
+
+const Onboarding: React.FC<Props> = ({ homeInformation }) => {
+  const [page, setPage] = useState('');
+  const [homeData, setHomeData] = useState(
+    homeInformation !== undefined ? homeInformation : homeInfo,
+  );
   const { setAppState } = useContext(AppContext);
+  const { setUserState } = useContext(UserContext);
 
   useEffect(() => {
+    const addHome = async () => {
+      let result: any;
+      if (homeInformation !== undefined) {
+        result = await API.graphql({
+          query: updateHome,
+          variables: { input: homeData },
+        });
+      } else {
+        result = await API.graphql({
+          query: createHome,
+          variables: { input: homeData },
+        });
+      }
+
+      await API.graphql({
+        query: createHomeOwner,
+        variables: {
+          input: {
+            homeID: result.data.createHome.id,
+            homeOwnerID: result.data.createHome.owner,
+          },
+        },
+      });
```

Comment:
Does the call at line 48 also run if the home information was updated (i.e. `updateHome` was called)? Since in that case, it would throw an error since `result.data.createHome` is undefined.

Reply:
Good catch! It is fixed in the latest commit.